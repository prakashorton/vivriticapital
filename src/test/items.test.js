import React from "react";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { mount } from "enzyme";
import Items from "../components/items";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });
describe("<Items />", () => {
  let wrapper = null;
  let props = {
    data: [
      {
        completed: false,
        title: "helloworld1",
        id: 1
      }
    ],
    onChange: jest.fn(),
    onEditTodo: jest.fn(),
    onDelete: jest.fn()
  };

  beforeEach(() => {
    wrapper = mount(<Items {...props} />);
  });

  it("renders correctly - snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders <Item /> components", () => {
    expect(wrapper).not.toBeNull();
  });

  it("renders check is items are rendered", () => {
    const uiItems = wrapper.find(".ant-list-items");
    const uiItem = uiItems.find(".ant-list-item");
    expect(uiItems.exists()).toBeTruthy();
    expect(uiItem.exists()).toBeTruthy();
    expect(uiItem.find(".ant-checkbox-input").exists()).toBeTruthy();
    expect(uiItem.find(".itemTitle").props().children).toBe(
      props.data[0].title
    );
  });

  it("renders check if onChange is triggered when checkbox is clicked", () => {
    expect(props.onChange).toHaveBeenCalledTimes(0);
    const uiItems = wrapper.find(".ant-list-items");
    const uiItem = uiItems.find(".ant-list-item");
    expect(uiItem.find(".ant-checkbox-input").exists()).toBeTruthy();
    uiItem
      .find(".ant-checkbox-input")
      .simulate("change", { target: { checked: true } });
    expect(props.onChange).toHaveBeenCalledTimes(1);
  });

  it("renders check if onDelete is triggered when user delete todo item", () => {
    expect(props.onDelete).toHaveBeenCalledTimes(0);
    const deleteButton = `.deleteBtn_1`;
    const uiItems = wrapper.find(".ant-list-items");
    expect(uiItems.find(deleteButton).exists()).toBeTruthy();
    uiItems
      .find(deleteButton)
      .first()
      .simulate("click");
    expect(props.onDelete).toHaveBeenCalledTimes(1);
  });

  it("renders edit button and save the edited change", () => {
    expect(props.onEditTodo).toHaveBeenCalledTimes(0);
    const editButton = `.editBtn_1`;
    const uiItems = wrapper.find(".ant-list-items");
    expect(uiItems.find(editButton).exists()).toBeTruthy();
    uiItems
      .find(editButton)
      .first()
      .simulate("click");
    wrapper.update();
    expect(wrapper.find(".saveBtn_1")).toBeTruthy();
    wrapper
      .find(".saveBtn_1")
      .first()
      .simulate("click");
    expect(props.onEditTodo).toHaveBeenCalledTimes(1);
  });
});
