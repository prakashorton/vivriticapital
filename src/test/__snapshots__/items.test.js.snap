// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Items /> renders correctly - snapshot 1`] = `
<Items
  data={
    Array [
      Object {
        "completed": false,
        "id": 1,
        "title": "helloworld1",
      },
    ]
  }
  onChange={[MockFunction]}
  onDelete={[MockFunction]}
  onEditTodo={[MockFunction]}
>
  <List
    bordered={true}
    dataSource={
      Array [
        Object {
          "completed": false,
          "id": 1,
          "title": "helloworld1",
        },
      ]
    }
    loading={false}
    pagination={false}
    renderItem={[Function]}
    size="large"
    split={true}
  >
    <div
      className="ant-list ant-list-lg ant-list-split ant-list-bordered"
    >
      <Spin
        size="default"
        spinning={false}
        wrapperClassName=""
      >
        <div
          className="ant-spin-nested-loading"
        >
          <div
            className="ant-spin-container"
            key="container"
          >
            <ul
              className="ant-list-items"
            >
              <Item
                key="list-item-0"
              >
                <li
                  className="ant-list-item"
                >
                  <div
                    className="pull-left"
                  >
                    <Checkbox
                      checked={false}
                      disabled={false}
                      indeterminate={false}
                      onChange={[Function]}
                    >
                      <label
                        className="ant-checkbox-wrapper"
                      >
                        <Checkbox
                          checked={false}
                          className=""
                          defaultChecked={false}
                          disabled={false}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          prefixCls="ant-checkbox"
                          style={Object {}}
                          type="checkbox"
                        >
                          <span
                            className="ant-checkbox"
                            style={Object {}}
                          >
                            <input
                              checked={false}
                              className="ant-checkbox-input"
                              disabled={false}
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              type="checkbox"
                            />
                            <span
                              className="ant-checkbox-inner"
                            />
                          </span>
                        </Checkbox>
                      </label>
                    </Checkbox>
                  </div>
                  <div
                    style={
                      Object {
                        "width": "75%",
                      }
                    }
                  >
                    <p
                      className="itemTitle"
                    >
                      helloworld1
                    </p>
                  </div>
                  <div
                    className="pull-right"
                  >
                    <Button
                      block={false}
                      className="editBtn_1"
                      ghost={false}
                      htmlType="button"
                      loading={false}
                      onClick={[Function]}
                    >
                      <Wave>
                        <button
                          className="ant-btn editBtn_1"
                          onClick={[Function]}
                          type="button"
                        >
                          <span
                            key=".0"
                          >
                            Edit
                          </span>
                        </button>
                      </Wave>
                    </Button>
                    <Button
                      block={false}
                      className="deleteBtn_1"
                      danger={true}
                      ghost={false}
                      htmlType="button"
                      loading={false}
                      onClick={[Function]}
                      type="primary"
                    >
                      <Wave>
                        <button
                          className="ant-btn deleteBtn_1 ant-btn-primary ant-btn-dangerous"
                          onClick={[Function]}
                          type="button"
                        >
                          <span
                            key=".0"
                          >
                            Delete
                          </span>
                        </button>
                      </Wave>
                    </Button>
                  </div>
                </li>
              </Item>
            </ul>
          </div>
        </div>
      </Spin>
    </div>
  </List>
</Items>
`;
